### DISCUSSION TITLE
Possible root cause of G513QM frequent AMBIENT mode lighting problem

### DISCUSSION THREAD

**Discussion by RAGE-HERO7** [seerge/g-helper#1906] [Q&A] [answered]
@seerge if the model Rog strix g15 G513QM exists with 2 types of keyboard (single zone/perkey Rgb) lighting types
this can be a problem for you right😕
but is it possible by adding
the number that represent specific model of the device like (G513QM-HQ403TS) to avoid conflict , i will show you three different G513QM and there specific model number.

![Screenshot_20240113_013227_Samsung Internet](https://github.com/seerge/g-helper/assets/154586512/2693c750-cf05-4f99-b499-406e00768f69)
![Screenshot_20240113_012623_Samsung Internet](https://github.com/seerge/g-helper/assets/154586512/260798c8-b87d-49da-9ccc-2a8dfa9ad3bd)
![Screenshot_20240113_012604_Samsung Internet](https://github.com/seerge/g-helper/assets/154586512/0aec45bd-e1f6-4585-9e13-2243e574cd67)

**Comment by seerge** [comment]
You can't read second part "HQ403TS"  anywhere. I can add a custom parameter for ``config.json`` where you can set perfered "mode" (per key / per zone)

**Reply by RAGE-HERO7** [reply]
ya that's a good idea, it's like a toggle right, please do it @seerge 🙏

**Reply by seerge** [reply]
You device is not even listed on asus website https://rog.asus.com/in/laptops/rog-strix/2021-rog-strix-g15-series/spec/  , where did you get it ? I can't find any open info saying it has per-key rgb

**Reply by seerge** [reply, marked as answer]
@RAGE-HERO7 try this build
[GHelper.zip](https://github.com/seerge/g-helper/files/13923832/GHelper.zip)

Add 
```
"per_key_rgb" : 1,
```

To the beginning of ``config.json``, and then start app

**Reply by RAGE-HERO7** [reply]
oh sorry i forgot to mention about it. idk what's going on , 1 year ago my device model disappeared from there website. with a normal search you can't find the model 😶
but if i log in to Asus website it will show up as registered device on my account with all the specifications lol.  
![20220508_161506](https://github.com/seerge/g-helper/assets/154586512/e32e830e-7dc3-4479-95eb-678f3d27ea8f)
![Screenshot_20240113_080832_Samsung Internet](https://github.com/seerge/g-helper/assets/154586512/664afe2e-e4ae-4f9e-8925-5b65924ce7e8)

**Reply by RAGE-HERO7** [reply]
@seerge where do i find this ' config.json ' though ?

**Reply by Kikimahe** [reply]
@RAGE-HERO7 Vous pouvez le trouver ici :
`C:\Users\yournameuser\AppData\Roaming\GHelper\config.json`  ;-)

**Reply by RAGE-HERO7** [reply]
Thank you :)

**Reply by RAGE-HERO7** [reply]
@seerge problem solved it is now working thank you. Hope it won't go away in next update🙂

**Comment by RAGE-HERO7** [comment]
@seerge 

i know this is a respected place only to share problems and opinions in a professional manner.

I did spammed comments(sorry about that) because the AMBIENT,HEATMAP problem emerged in the new version, even after you got it fixed on this GHELPER exe that you send me. https://github.com/seerge/g-helper/discussions/1781#discussioncomment-7937343
if i set not working AMBIENT or HEATMAP and restart laptop it gets stuck in booting and shows only republic of gamers logo for 15 minutes until it restarts normally ☹️
 
i am not paying you to fix the app for me.
your time is important same as everyone and i regret to waste it further more :)
so i decided to never post about this small lighting problem again , it looks like this AMBIENT,HEATMAP issue still gonna emerge after some other G513QM owner ask you to fix the lighting and that gonna reload this problem on new update, so i decided to live with not working 2 light modes :) 

It is completely understandable you don't have all kinds of Asus laptops to test and fix problems for everyone and you almost do it perfectly and fixes the problem eventually within your knowledge and limited resources, you are awsome bro 🙏

about me
i am a busy guy running around, studying for exams and helping with reconstruction of my parents home and taking care of there health problems.so i can't be with my device everyday :(
But i am still gonna use GHELPER till my Laptop dies and recommended the app to my friends.

The Simplest app ever made by 1 chad guy
The Best❤️‍🔥
Thank you🫡

**Reply by seerge** [reply]
@RAGE-HERO7 hello, I don't think that backlight mode have any relationship to your booting. It's just a color set via usb.

I'm glad that config option added helped you.
